### PR TITLE
【AutoParallel】Add some options for dist.Strategy

### PR DESCRIPTION
**Pull request by heavyrain-lzy**
### PR types

Others
### PR changes

Others
### Description

Pcard-76459
Add other options for `dist.Strategy`.

### CODE REVIEW

**Comment by heavyrain-lzy** [comment]
用法参见：https://github.com/PaddlePaddle/PaddleNLP/pull/8114

File: python/paddle/distributed/auto_parallel/api.py [hunk 1]
```diff
@@ -1405,6 +1405,51 @@ def __init__(self, config=None):
...
+        config_dict = self._config_dict.get(
+            auto_strategy.constants.DP_OPTIMIZATION, None
+        )
+        self._dp_optimization = auto_strategy.DPOptimizationConfig(config_dict)
+        config_dict = self._config_dict.get(
+            auto_strategy.constants.SP_OPTIMIZATION, None
+        )
+        self._sp_optimization = auto_strategy.SPOptimizationConfig(config_dict)
+
+    def _convert_to_unify_strategy(self, auto_stragety):
```
Comment (zhiqiu) [inline python/paddle/distributed/auto_parallel/api.py]: -> _from_legacy_strategy()

Comment (heavyrain-lzy) [reply python/paddle/distributed/auto_parallel/api.py]: Done. Thanks~

**Review by zhiqiu — APPROVED** [review]
LGTM

**Pull request merged**